### PR TITLE
Get native handles for TextureResource and BufferResource

### PR DESCRIPTION
**Pull request by lucy96chen**
Changes:
- Added `getNativeHandle()` to `ITextureResource` and `IBufferResource`
- Implemented `getNativeHandle()` for Vulkan and D3D12 as well as in `renderer_shared.cpp`
- Added two unit tests testing the above functionality
- Made small modifications to previously added unit tests

@csyonghe

### CODE REVIEW

File: tools/gfx-unit-test/existing-device-handle-test.cpp [hunk 1]
```diff
@@ -100,7 +100,7 @@ namespace gfx_test
     {
         if ((api & context->enabledApis) == 0)
```
Comment (csyonghe) [inline tools/gfx-unit-test/existing-device-handle-test.cpp:101]: You can use the new runTestImpl function to eliminate all the device creation logic here. See the current compute-smoke test for an example.

**Review by csyonghe — APPROVED** [review]
Looks good, you may consider simplify the code by making use of the latest runTestImpl function.

File: tools/gfx-unit-test/existing-device-handle-test.cpp (hunk 1 above)
Comment (lucy96chen) [reply tools/gfx-unit-test/existing-device-handle-test.cpp:101]: I'm not seeing that function anywhere. Where is it supposed to be?

Comment (csyonghe) [reply tools/gfx-unit-test/existing-device-handle-test.cpp:101]: Ah, never mind, I just realized that it is still in my local branch and I haven't committed it to master yet.

Comment (lucy96chen) [reply tools/gfx-unit-test/existing-device-handle-test.cpp:101]: Ah, okay. I'll make this change in a later PR once it is then.

**Pull request merged**